Quartz port of map-app

zinfin opened the pull request:
@doneill I've ported almost all of the functionality to Quartz in this first pass.  I tried to maintain the original structure of the app and focused on a pure port.  The route task pieces are in place but not working as expected.  Can you look over what I have so far?

doneill commented on maps-app/src/main/java/com/esri/android/mapsapp/MapFragment.java:
```diff
@@ -96,8 +98,10 @@
 import com.esri.arcgisruntime.mapping.view.LocationDisplay;
 import com.esri.arcgisruntime.mapping.view.MapView;
 import com.esri.arcgisruntime.mapping.view.WrapAroundMode;
+import com.esri.arcgisruntime.portal.LicenseInfo;
```
This import is unused, please remove.

doneill commented on maps-app/src/main/java/com/esri/android/mapsapp/MapFragment.java:
```diff
@@ -96,8 +98,10 @@
+import com.esri.arcgisruntime.portal.LicenseInfo;
 import com.esri.arcgisruntime.portal.Portal;
 import com.esri.arcgisruntime.portal.PortalItem;
+import com.esri.arcgisruntime.security.UserCredential;
```
This import is unused, please remove.

doneill commented on maps-app/src/main/java/com/esri/android/mapsapp/MapFragment.java:
```diff
@@ -79,6 +79,8 @@
 import com.esri.android.mapsapp.util.TaskExecutor;
 import com.esri.arcgisruntime.ArcGISRuntimeEnvironment;
+import com.esri.arcgisruntime.LicenseLevel;
+import com.esri.arcgisruntime.LicenseResult;
```
The `ArcGISRuntimeEnvironment`, `LicenseLevel`, & `LicenseResult` imports are unused, please remove.

doneill commented:
Some general comments, which may or may not have been introduced in this PR 😄 : 

- Return value of `MapFragment.onGetRoute()` method is never used, can you confirm this and refactor return value to `void`?  
- Replace explicit type argument in `MapFragment.getSuggestions()` with diamond `<>`. 
- The value of `locationChanged` in `MapFragment.startLocationTracking()` is always `false`.  I think we can remove the conditional as the value `true` assigned is never used? 
- The collection `suggestedAddresses` gets updated `MapFragment.getSuggestions()` but are never used?  Can we safely remove?

doneill commented:
@zinfin I have some comments for your review.  Still working through some more though ...

zinfin commented:
@doneill I've completed the fixes we discussed last week:

- Regarding `MapFragment.onGetRoute()`, I looked into refactoring and returning value of `void` but we're implementing the 'RoutingDialogListener' interface from the 'RoutingDialogFragment' class has the callback method `boolean onGetRoute(String startPoint, String endPoint)` so I've kept it the same.
- The `MapFragment.getSuggestions()` has bee replaced with diamond `<>`.
- I've removed the `MapFragment.startLocationTracking()` method altogether, handling location tracking in the `setMapView` method.
- I've removed `suggestedAddresses` as well as other un-used class variables.
- After talking with Justin and Will this week about routing in the app, they wanted the app to use the routing service associated with the user's ArcGIS Online account so I've added that logic.  I did bring up the issue that you and I discussed about the onus of credit use falling to the user.  We talked about getting proxies to work which would put the onus back on the developer (right?).  I'm still exploring this.
- Previously, we called `mMapView.restoreState(mMapViewState)` in the `onResume method`.  How is state managed through the activity or fragment lifecycle?  Do we need to do anything explicit in the `onPause` or `onResume` methods?
- I've fixed the behavior of the FAB, it now toggles between AutoPan modes (OFF and COMPASS).  Is this the behavior we were looking for?
- Location display is turned on in the `setMapView`.  The `setMapView` is called `onCreateView` and every time a web map is loaded.  Do all of the set up things in the map view have to be done every time we change the map?
- I tried to fix the compass behavior but to no avail, it's still not working correctly.
- I've resized the PNGs in the asset folder and added a few more for routing.  Are they the right size or should I make them smaller?

doneill commented:
Thanks @zinfin I will review!

doneill commented:
@zinfin I started a cursory review of the UI/UX and couple of things:  

- The compass mode is not working, tapping the compass button does not orientate the map
- Route is requiring a login.  Do not want to commit to that user experience yet, still want to seek a pattern to support not logging in to perform a route.  

More to come ...

zinfin commented:
@doneill 

- I can't figure out how to fix the compass and need some pointers to get me started.

- Would provisioning a proxy be a solution for the routing service?

zinfin commented:
@doneill When thinking about the documentation we'll be providing about the port from 10.2.7 to quartz-beta it seems like we want to center any discussions on comparing the differences in handling 1) the map view, 2) location display, 3) geocoding and routing (covering loadables and futures), and 4) authentication.  When starting to write this doc, I'm trying to understand the motivation for creating the MapView inline with 10.2.7 instead of relying on markup in the layout file.  Can you remind me?

doneill commented:
Yes, the reason for the `MapView` API pattern as opposed to the XML `View` pattern is that the Portal API only supports the API pattern when switching a Basemap in a `MapView`.

doneill commented on maps-app/src/androidTest/java/com/esri/android/maps/app/MappsAppActivityTest.java:
```diff
@@ -0,0 +1,129 @@
+
+/**
+ * Created by sand8529 on 4/26/16.
+ */
```
Remove default file template and replace with Test Class description

doneill commented on maps-app/src/androidTest/java/com/esri/android/maps/app/MappsAppActivityTest.java:
```diff
@@ -0,0 +1,129 @@
+package com.esri.android.maps.app;
```
Add copyright

doneill commented:
> Previously, we called `mMapView.restoreState(mMapViewState)` in the `onResume method`.  How is state managed through the activity or fragment lifecycle?  Do we need to do anything explicit in the `onPause` or `onResume` methods?

We have not implemented those helper methods as of yet, but `MapView` extends `ViewGroup` so you can override `ViewGroup.onSaveInstanceState` and `ViewGroup.onRestoreInstanceState()`

willcrick commented:
> the reason for the MapView API pattern as opposed to the XML View pattern is that the Portal API only supports the API pattern when switching a Basemap in a MapView.

The basemap is switched on the Map, rather than the MapView, so it shouldn't matter how the MapView is constructed (unless i am missing something).

doneill commented:
@willcrick You are referencing quartz, the context of question is design from 10.2x.  You are correct in pattern moving forward.

doneill commented:
@zinfin Looks good to merge, I will merge and close.
